File: 001/problem_001.cpp
#include <iostream>

using namespace std;

int sumtoN(const int n)
{
    return n*(n+1)/2;
}

int sumMultiples(const int limit, const int a)
{
    return sumtoN((limit -1)/a)*a;
}

int main()
{
    const int N=1000;
    cout<<"result = "<<(sumMultiples(N,3)+sumMultiples(N, 5)-sumMultiples(N, 15))<<endl;
    return 0;
}

File: 001/problem_001.py
# -*- coding: utf-8 -*-

# answer is 233168

def sum1toN(n):
   return n * (n + 1) / 2

def sumMultiples(limit, a):
   return sum1toN((limit - 1) / a) * a

if __name__ == '__main__':
    N=1000
    result = sumMultiples(N, 3) + sumMultiples(N, 5) - sumMultiples(N, 15)
    print("result = %d"% result)
